docs/source/developers/release.rst

```diff
@@ -87,12 +87,49 @@ These are the different steps that are required to create a release candidate.
     # Curate the release
     # The end of the generated report shows the jira tickets with wrong version number assigned.
     archery release curate <version>
-    
+
+
+Creating a Release Candidate
+============================
+
+These are the different steps that are required to create a release candidate.
+
+The initial Release Candidate branch is merged from master. Follow up Release
+Candidates will be created from a maintenance branch.
+
+We have implemented a Feature Freeze policy between Release Candidates.
+This means that, in general, we will only add fixes between Release Candidates.
+Only specific features, if there is community consensus, will be merged to further Release Candidates.
+
+Create or update the corresponding maintenance branches:
+
+.. tab-set::
+
+   .. tab-item:: Initial Release Candidate
+
+      .. code-block::
+
+            # From an up to date master branch execute
+            archery release --jira-cache /tmp/jiracache cherry-pick X.Y.Z --execute
+
+   .. tab-item:: Follow up Release Candidates
+
+      .. code-block::
+
+            # First run on dry-mode to see what are the commits that will be cherry-pick.
+            # If there are commits that we don't want to get applied ensure the version on
+            # JIRA is set to the following release.
+            archery release --jira-cache /tmp/jiracache cherry-pick X.Y.Z --continue
+            # Update the maintenance branch with the previous commits
+            archery release --jira-cache /tmp/jiracache cherry-pick X.Y.Z --continue --execute
+
+Create the rest of the Release branches.
+
+.. code-block::
+
     # Checkout release branch
     # Use master for major releases
-    git checkout -b release-4.0.0 master
```

Comment:
I don't quite understand this change?

Reply:
My idea is that as we will always be using the maintenance branch we don't require to branch from master anymore and for a follow up release candidate we want to create the release branch from the maintenance branch instead of master.

Reply:
But where/when is `maint-X.Y.Z` created? e.g. there is no maint-8.0.0

> for a follow up release candidate we want to create the release branch from the maintenance branch instead of master.

👍

Reply:
The idea is that prior to this you are going to generate the maintenance branch via the command:
```
archery release --jira-cache /tmp/jiracache cherry-pick X.Y.Z --execute
```
This generates that branch from master on the first release and updates the maintenance branch for following release candidates via:
```
archery release --jira-cache /tmp/jiracache cherry-pick X.Y.Z --continue --execute
```
If it was not clear I can add a comment specifying that a maintenance branch should be created at this point?

Reply:
New commit added to make it more clear, let me know what you think: https://github.com/apache/arrow/pull/13308/commits/adf1a4501f4b2afaaad4fb41fc99c6eb348dad25